torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -262,52 +280,78 @@ def get_aten_name(
         raise diagnostics.RuntimeErrorWithDiagnostic(diagnostic)
 
     @_beartype.beartype
+    @diagnostics.diagnose_call(
+        diagnostics.rules.find_operator_overloads_in_onnx_registry,
+        diagnostic_message_formatter=_find_operator_overloads_in_onnx_registry_disagnostic_message_formatter,
+    )
     def get_function_overloads(
         self,
         node: torch.fx.Node,
-        aten_name: str,
+        internal_opname: registration.OpName,
         diagnostic_context: diagnostics.DiagnosticContext,
-    ) -> Set[Union["onnxscript.OnnxFunction", "onnxscript.TracedOnnxFunction"]]:
+    ) -> List[registration.SymbolicFunction]:
         """Get the function overloads from the registry.
 
         Args:
             node: The node to get the function overloads for.
-            aten_name: The aten name of the node.
+            internal_opname: registration.OpName.
             diagnostic_context: The diagnostic context to use for reporting errors.
 
         Returns:
-            Set of function overloads.
+            The list contains SymbolicFunctions, starting with the default ones and
+            followed by any custom ones.
         """
-        function_group = None
 
-        if self.onnx_registry.is_registered_op(aten_name, self.opset_version):
-            function_group = self.onnx_registry.get_function_group(aten_name)
+        # NOTE: If the ATen/Custom operators are nt registered, the group will be None.
+        # And non-registerd ATen/Custom operators will trigger error in the next step.
+        function_group: Optional[List[registration.SymbolicFunction]] = None
 
-        # Fall back to overloadpacket name: eg: aten.add.Tensor -> aten::add
-        elif hasattr(
-            node.target, "overloadpacket"
-        ) and self.onnx_registry.is_registered_op(
-            node.target.overloadpacket._qualified_op_name,  # type: ignore[union-attr]
-            self.opset_version,
+        if self.onnx_registry.is_registered_op(
+            domain=internal_opname.domain,
+            op_name=internal_opname.op_name,
+            overload=internal_opname.overload,
         ):
-            function_group = self.onnx_registry.get_function_group(
-                node.target.overloadpacket._qualified_op_name  # type: ignore[union-attr]
+            function_group = self.onnx_registry.get_functions(
+                domain=internal_opname.domain,
+                op_name=internal_opname.op_name,
+                overload=internal_opname.overload,
             )
 
-        if function_group is not None:
-            # TODO(titaiwang): dispatch opset version.
-            dispatched_version = _dispatch_opset_version(
-                self.opset_version, function_group.support_opset()
+        # NOTE: Fall back to default overload if the ONNX registry doesn't have the overload.
```

Comment:
@justinchuby I found that torchlib really needs default fall back system. For example, aten::add.Tensor falls back to aten::add to find registry. Do you think this is something we need to address in torchlib side? Right now, I put a warning in SARIF if it's overload not found in the registry, but default is.

Reply:
`aten::add` is the default we fall back on. Is there something we are missing?

Reply:
It's just dispatching to it's overload is preferred to falling back. I am saying are we safe relying on fall back? Or we will address that: add aten::add.Tensor into torchlib eventually?

Reply:
I think this deserves a discussion. https://github.com/microsoft/onnxscript/issues/828